server/app/acs_enterprise/utils/dynamodb_utils.py

```
@@ -29,174 +28,196 @@ class ConcurrencyController:
+                new_count = int(response['Attributes']['active_calls']['N'])
+                logger.info(f"✅ Acquired concurrency slot for call {call_id} (active: {new_count}/{self.max_concurrent})")
+                return True
 
         except ClientError as e:
             if e.response['Error']['Code'] == 'ConditionalCheckFailedException':
                 # At capacity
                 logger.warning(f"⚠️ Concurrency limit reached ({self.max_concurrent} active calls)")
                 return False
             else:
-                logger.error(f"❌ DynamoDB error acquiring slot: {e}")
+                logger.error(f"❌ DynamoDB error acquiring slot for call {call_id}: {e}")
```

Ruff (0.15.1)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)